sdk/identity/Azure.Identity/src/AzureArcManagedIdentitySource.cs

```diff
@@ -51,6 +51,7 @@ protected override Request CreateRequest(string[] scopes)
             // arc MI endpoint doesn't support user assigned identities so if client id was specified throw AuthenticationFailedException
             if (!string.IsNullOrEmpty(_clientId))
```

Comment:
Is it possible to pass a ResourceID to this source?

Reply:
In theory any of them are subject to the option if it was specified. We happen not to explicitly do anything with it here, but we should warn if it was sent in the options either way. I'll add the additional check.